forem/forem
# Removed service worker

## What type of PR is this? (check all applicable)


## Description

This removes all of our service worker functionality with a self-destructing service worker which appears to be the best approach for unregistering a service worker. See https://github.com/NekR/self-destroying-sw#how-to-use.

## Related Tickets & Documents

#12899
#12631
#12455
#12000
#11606 
#11533
#11488
#11270
#10293
#10173
#9657 
#9642 
#9509
#9254
#9228
#8729
#7339
#7081
#6497

## QA Instructions, Screenshots, Recordings

**Test this in Chrome, Firefox, and Safari. Note in Safari, there may be no service worker registered due to the work done with the service-companion.js file. In that case, ensure that no errors are thrown in the console of the web dev tools in Safari.**

1. Pull down latest of the main branch and deploy the site locally. Navigate to the home page and validate that the service worker is running by opening the developer tools and navigating to the developer tools.

**Chrome**

![image](https://user-images.githubusercontent.com/833231/110863755-25bcda80-828f-11eb-98dd-38e9c1a12ce9.png)

**Firefox**

![image](https://user-images.githubusercontent.com/833231/110955507-ade6c280-8317-11eb-8e3b-d306f4226a0c.png)

**Safari**

![image](https://user-images.githubusercontent.com/833231/110955743-e7b7c900-8317-11eb-95e7-91935d25cf9d.png)

2. Pull down the branch for this PR and deploy it locally. Refresh the browser window that is currently open and validate that the service worker gets unregistered.

**Chrome**

![image](https://user-images.githubusercontent.com/833231/110863832-42591280-828f-11eb-8563-bb959a81d101.png)

**Firefox**

![image](https://user-images.githubusercontent.com/833231/110955675-d66ebc80-8317-11eb-99cc-7c20cafcb94c.png)

**Safari**

![image](https://user-images.githubusercontent.com/833231/110955754-e9818c80-8317-11eb-9dc8-095c7f6a2eb9.png)

### UI accessibility concerns?

N/A this is front-end infrastructure

## Added tests?

- [x] No, and this is why: I've removed tests related to the service worker, so existing tests should continue to pass.

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [x] @mstruve will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Samuel L. Jackson in Jurassic Park saying "Hold on to your butts"](https://media.giphy.com/media/2Z3lgZOhISkYU/giphy.gif)
